### PR TITLE
Add QueryJob#query_plan

### PR DESCRIPTION
**Pull request by quartzmo**
Expose `Job.statistics.query.queryPlan`.

Add `QueryJob::Stage` and `QueryJob::Step` classes.

[refs #1634]

### CODE REVIEW

File: google-cloud-bigquery/lib/google/cloud/bigquery/query_job.rb [hunk 1]
```diff
@@ -106,6 +106,18 @@ def bytes_processed
         end
 
         ##
+        # Describes the execution plan for the query.
+        #
+        # @return [Array<Google::Cloud::Bigquery::QueryJob::Stage>]
+        #
+        def query_plan
+          return nil unless @gapi.statistics.query.query_plan
```
Comment (blowmage) [inline google-cloud-bigquery/lib/google/cloud/bigquery/query_job.rb]: What if `@gapi.statistics` is nil? Or `@gapi.statistics.query` is nil?

Comment (quartzmo) [inline google-cloud-bigquery/lib/google/cloud/bigquery/query_job.rb]: The existing readers already assume these will always be there. And I think that's correct wrt the API. However, `query_plan` is only available for 7 days per the docs, and therefore may be `nil`.

```ruby
##
        # Checks if the query results are from the query cache.
        def cache_hit?
          @gapi.statistics.query.cache_hit
        end
```

Comment (blowmage) [inline google-cloud-bigquery/lib/google/cloud/bigquery/query_job.rb]: That's right. I think we observed that a query job always has `statistics.query`, which is why it isn't guarded anywhere else.

File: google-cloud-bigquery/acceptance/bigquery/table_test.rb [hunk 2]
```diff
@@ -253,6 +253,36 @@
     query_job.data.class.must_equal Google::Cloud::Bigquery::Data
     query_job.data.total.wont_be :nil?
 
+    # Query Job - Statistics Query Plan
+    query_job.query_plan.wont_be_nil
+    query_job.query_plan.must_be_kind_of Array
+    query_job.query_plan.wont_be :empty?
+    stage = query_job.query_plan.first
```
Comment (quartzmo) [inline google-cloud-bigquery/acceptance/bigquery/table_test.rb]: @blowmage Are you OK with `#query_plan` returning an array, even though it sounds like a singular noun? Pretty misleading if you ask me, but I'm just sticking to the underlying API.

Comment (blowmage) [inline google-cloud-bigquery/acceptance/bigquery/table_test.rb]: Yeah, I'd stick with the name.

**Pull request merged**